lib/urlwatch/filters.py

```diff
@@ -330,6 +331,46 @@ def filter(self, data, subfilter):
                          method=method, options=options)
 
 
+class Csv2TextFilter(FilterBase):
+    """Convert CSV to plaintext"""
+
+    __kind__ = 'csv2text'
+
+    __supported_subfilters__ = {
+            'format_message': 'A format string with the headers that will be outputted for each csv '
+                                                         'line (header will be lower-cased)',
+            'ignore_header': 'If your format string is number based, but the CSV has headers, '
+                                                       'this flag will force ignoring the header.',
+        }
+
+    __default_subfilter__ = 'format_message'
+
+    def filter(self, data, subfilter):
+        has_header = csv.Sniffer().has_header(data)
```

Comment:
Should it be possible to supply a subfilter, so that:

1. If the subfilter `has_header` is `True` or `False`, it overrides the `has_header`
2. If there is no `has_header` set in the subfilter, `csv.Sniffer()` can be used

Reply:
To make sure I understand, you are suggesting that if `has_header` is False we ignore sniffer result and just include the header as part of the data?
To me it seems like allowing someone to shoot themselves in the foot, but if you feel that this customization is needed, then sure, I'll add.

Reply:
The thing is, the sniffer might be wrong, so it makes sense to force-disable it. If there's no way to disable it, and if it "guesses" wrong, the filter will be unusable for certain data.